### PR TITLE
doc: Update API documentation

### PR DESCRIPTION
**Pull request by crandmck**
Review doc comments for Reader.

## Changes in this pull request

Edits to doc comments in Rust code for clarity and completeness.

TO DO:
- Change all references to errors to be to the [`Error`] enum

## Checklist
- [x] This PR represents a single feature, fix, or change.
- [x] All applicable changes have been documented.
- [ ] Any `TO DO` items (or similar) have been entered as GitHub issues and the link to that issue has been included in a comment.

### CODE REVIEW

File: sdk/src/builder.rs [hunk 1]
```diff
@@ -112,6 +115,8 @@ fn default_vec<T>() -> Vec<T> {
     Vec::new()
 }
 
+/// Assertion data is almost always a JSON structure, but this allows for
```
Comment (gpeacock) [inline sdk/src/builder.rs:118]: This comment applies to the Cbor variant of this enum. I think all we need to say here is that.
This allows the assertion to be expressed as CBOR or JSON. 
The default is CBOR unless you specify that an assertion should be JSON.

File: sdk/src/builder.rs [hunk 2]
```diff
@@ -80,6 +79,8 @@ pub struct ManifestDefinition {
     #[serde(default = "default_instance_id")]
     pub instance_id: String,
 
+    /// An optional ResourceRef to an image.  Must be available when the manifest is signed.
```
Comment (gpeacock) [inline sdk/src/builder.rs:82]: This needs to say something about being a thumbnail image that represents the asset.

File: sdk/src/ingredient.rs [hunk 3]
```diff
@@ -544,15 +544,15 @@ impl Ingredient {
...
+    /// Generates an `Ingredient` from a stream, including XMP info.
     pub fn from_stream_info<F, S>(stream: &mut dyn CAIRead, format: F, title: S) -> Self
     where
         F: Into<String>,
         S: Into<String>,
     {
         let format = format.into();
 
-        // try to get xmp info, if this fails all XmpInfo fields will be None
+        // try to get xmp info, if this fails all XmpInfo fields will be None.
```
Comment (gpeacock) [inline sdk/src/ingredient.rs:555]: Capitalize the T at the beginning.

**Review by gpeacock — CHANGES_REQUESTED** [review]
I requested some changes but otherwise good to go.

File: sdk/src/builder.rs (hunk 1 above)
Comment (crandmck) [reply sdk/src/builder.rs]: Addressed.

File: sdk/src/builder.rs (hunk 2 above)
Comment (crandmck) [reply sdk/src/builder.rs]: Addressed.

File: sdk/src/ingredient.rs (hunk 3 above)
Comment (crandmck) [reply sdk/src/ingredient.rs]: Addressed, although this is not a doc comment.

**Comment by crandmck** [comment]
Addressed your comments.

**Review by gpeacock — APPROVED** [review]

**Pull request merged**